pkg/volume/csi/csi_plugin_test.go

```diff
@@ -652,23 +663,29 @@ func TestPluginNewMounterWithInline(t *testing.T) {
 			shouldFail: true,
 		},
 		{
-			name:       "mounter with persistent volume source",
-			spec:       volume.NewSpecFromPersistentVolume(makeTestPV("test-pv1", 20, testDriver, testVol), true),
-			podUID:     types.UID(fmt.Sprintf("%08X", rand.Uint64())),
-			namespace:  "test-ns1",
-			driverMode: persistentDriverMode,
+			name:          "mounter with persistent volume source",
+			spec:          volume.NewSpecFromPersistentVolume(makeTestPV("test-pv1", 20, testDriver, testVol), true),
+			podUID:        types.UID(fmt.Sprintf("%08X", rand.Uint64())),
+			namespace:     "test-ns1",
+			csiVolumeMode: persistentVolumeMode,
 		},
 		{
-			name:       "mounter with volume source",
-			spec:       volume.NewSpecFromVolume(makeTestVol("test-vol1", testDriver)),
-			podUID:     types.UID(fmt.Sprintf("%08X", rand.Uint64())),
-			namespace:  "test-ns2",
-			driverMode: ephemeralDriverMode,
+			name:          "mounter with volume source",
+			spec:          volume.NewSpecFromVolume(makeTestVol("test-vol1", testDriver)),
+			podUID:        types.UID(fmt.Sprintf("%08X", rand.Uint64())),
+			namespace:     "test-ns2",
+			csiVolumeMode: ephemeralVolumeMode,
 		},
 	}
 
 	for _, test := range tests {
-		plug, tmpDir := newTestPlugin(t, nil)
+		driverMode := storagev1beta1.PersistentDriverMode
+		if test.csiVolumeMode == ephemeralVolumeMode {
+			driverMode = storagev1beta1.EphemeralDriverMode
+		}
+		driver := getTestCSIDriver(testDriver, nil, nil, &driverMode)
+		fakeClient := fakeclient.NewSimpleClientset(driver)
+		plug, tmpDir := newTestPlugin(t, fakeClient)
 		defer os.RemoveAll(tmpDir)
 
```

Comment:
since we use defer in foreach，using t.Run is better

Reply:
Right, good catch. Other tests already did that and I missed that this one doesn't. Will fix it.

Reply:
Actually there was a `t.Run`, I just moved the code into the wrong spot. Fixed.